### PR TITLE
Added: StandardWellAssemble

### PR DESCRIPTION
**Pull request by akva2**
This continues https://github.com/OPM/opm-simulators/pull/4268

Here we move assemble functions to a separate class, and make the equation system private in StandardWellEquations.
A friend declaration is used to allow the assembler (mutable) access to the matrices/vectors.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — COMMENTED** [review]
This looks good for the most part. I just noticed a couple of issues that I'd like to see your response to before merging.

File: opm/simulators/wells/StandardWellAssemble.hpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    //! \brief Assemble a source term.
+    template<class EvalWell>
+    void assembleSourceEq(const EvalWell& resWell_loc,
+                          const int componentIdx,
+                          const int numWellEq,
+                          StandardWellEquations<Scalar,Indices::numEq>& eqns) const;
+
+
+private:
+    const WellInterfaceFluidSystem<FluidSystem> well_; //!< Reference to well
```
[the remark this answers is not in the document]
Comment (akva2) [reply opm/simulators/wells/StandardWellAssemble.hpp]: nice catch! this is indeed supposed to be a reference.

File: opm/simulators/wells/StandardWellAssemble.cpp [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+                  const int Bhp,
+                  StandardWellEquations<Scalar,Indices::numEq>& eqns1,
+                  DeferredLogger& deferred_logger) const
+{
+    static constexpr int Water = BlackoilPhases::Aqua;
+    static constexpr int Oil = BlackoilPhases::Liquid;
+    static constexpr int Gas = BlackoilPhases::Vapour;
+    EvalWell control_eq(numWellEq + Indices::numEq, 0.0);
+
+    StandardWellEquationAccess eqns(eqns1);
```
[the remark this answers is not in the document]
Comment (akva2) [reply opm/simulators/wells/StandardWellAssemble.cpp]: I can.

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks for the updates.  This looks good to me now and I'll merge into master.

**Pull request merged**